src/DTO/Bandit/IContextAttributes.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Eppo\DTO\Bandit;
+
+/**
+ * A set of attributes for a given context ($key))
+ */
+interface IContextAttributes
+{
+    public function getAttributes(): AttributeSet;
+
+    public function getKey(): string;
```

Comment:
What do we need `$key` for? Each attribute should have it's own key.

Reply:
`ContextAttributes` pairs a string `key` with an `AttributeSet` in order to keep SubjectIDs and ActionKeys closer to their attributes when dealing with groups of same. It's a convenience layer that alleviates some awkwardness with having to access both key and value when manipulating arrays.

I've updated the docs on what exactly the `IContextAttributes` interface is intended for/what problem it solves. Hopefully that makes it clear why the added layer on top of `AttributeSet`